kong/plugins/moesif/log.lua

```diff
@@ -210,43 +219,46 @@ local function send_events_batch(premature)
     return
   end
 
+  local send_events_socket = ngx.socket.tcp()
+  send_events_socket:settimeout(1000)
+
   local batch_events = {}
   repeat
     for key, queue in pairs(queue_hashes) do
-      if #queue > 0 then
+      if #queue > 0 and ((socket.gettime()*1000 - start_time) <= 1000) then
```

Comment:
@dgilling, this is the other safeguard, we added yesterday. Earlier behavior is , when timer callback called, as long as there is events in queue, we keep on sending to moesif. I think timer callbacks has to executed as quick as possible. So put a check here..that this call back before sending the next batch, we check if time spent in the callback is > 1 second..then we exit the callback..we ll send remaining events next time when call back is called. In a ideal world, this limit on 1 second shouldn't be bad..however this throttles number of events sent to moesif

Reply:
lgtm

Reply:
max_callback_time_spent 2 seconds default